src/test/scala/org/ergoplatform/nodeView/state/UtxoStateSpecification.scala

```diff
@@ -3,23 +3,87 @@ package org.ergoplatform.nodeView.state
 import java.util.concurrent.Executors
 
 import io.iohk.iodb.ByteArrayWrapper
+import org.ergoplatform.mining._
 import org.ergoplatform.modifiers.ErgoFullBlock
 import org.ergoplatform.modifiers.history.{ADProofs, BlockTransactions, Extension, Header}
-import org.ergoplatform.modifiers.mempool.ErgoTransaction
+import org.ergoplatform.modifiers.mempool.{ErgoTransaction, UnsignedErgoTransaction}
 import org.ergoplatform.nodeView.history.ErgoHistory
 import org.ergoplatform.nodeView.state.wrapped.WrappedUtxoState
 import org.ergoplatform.utils.ErgoPropertyTest
 import org.ergoplatform.utils.generators.ErgoTransactionGenerators
 import org.ergoplatform.{ErgoBox, ErgoBoxCandidate, Input}
 import scorex.core._
+import scorex.util.encode.Base16
 import sigmastate.Values
+import sigmastate.Values.ByteArrayConstant
+import sigmastate.basics.DLogProtocol.{DLogProverInput, ProveDlog}
 
 import scala.concurrent.{ExecutionContext, ExecutionContextExecutor, Future}
 import scala.util.{Random, Try}
 
 
 class UtxoStateSpecification extends ErgoPropertyTest with ErgoTransactionGenerators {
 
+  property("Founders should be able to spend genesis founders box") {
+    var (us, bh) = createUtxoState()
+    val foundersBox = genesisBoxes.last
+    var height: Int = ErgoHistory.GenesisHeight
+
+    val settingsPks = settings.chainSettings.foundersPubkeys
+      .map(str => pkFromBytes(Base16.decode(str).get))
+      .map(pk => ProveDlog(pk))
+    settingsPks.count(p => defaultProver.dlogPubkeys.contains(p)) shouldBe 2
+
+    forAll(defaultHeaderGen) { header =>
+      val rewardPk = (new DLogProverInput(BigInt(header.height).bigInteger)).publicImage
+
+      val t = validTransactionsFromBoxHolder(bh, new Random(height))
+      val txs = t._1
+      bh = t._2
```

Comment:
@catena2w What is this?

Reply:
Is there a better way to do this? `validTransactionsFromBoxHolder` return tuple, we should update `bh` variable to the second element of the tuple